tests/test_modeling_utils.py

```diff
@@ -1850,3 +1851,100 @@ def test_not_available_sdpa(self):
             )
 
         self.assertTrue("PyTorch SDPA requirements in Transformers are not met" in str(cm.exception))
+
+
+@require_torch
+@slow
+class Mask4DTest(unittest.TestCase):
+    def setUp(self):
+        self.device = torch.device("cuda:0")
+        model_name = "JackFram/llama-160m"  # small Llama-like model from FlexFlow
+        self.tokenizer = AutoTokenizer.from_pretrained(model_name)
+        self.model = AutoModelForCausalLM.from_pretrained(model_name, torch_dtype=torch.float32).to(self.device)
```

Comment:
```suggestion
        self.model = AutoModelForCausalLM.from_pretrained(model_name, torch_dtype=torch.float16).to(self.device)
```
the smaller the better for our CI

Reply:
I observed that fp16 tests are more noisy, so what I did is:
- retained fp32 testsm but used even smaller model
- added fp16 test with relaxed tolerances
- added fp16 testing option for the top tokens order.